docs/2.8.0/docs/tools/assistant-build.rst

```diff
@@ -15,21 +15,502 @@ To compile your Daml source code into a Daml archive (a ``.dar`` file), run::
+
+Build without multi-package
+-----------------------------------
+
+As context for the multi-package illustrations in later sections,
+here's an example of how to use ``daml build`` without the multi-package 
+feature:
+
+.. code::
+
+   > # ... make changes to Model ...
+   > cd ./package-Model/  # Navigate to package-Model
+   > daml build           # Build Model.dar
+   > cd ../package-Logic/ # Navigate to package-Logic
+   > daml build           # Build Logic.dar
+
+Multi-package configuration
+===========================
+
+To use multi-package builds, create a ``multi-package.yaml`` configuration file. This file serves as a register of the interdependent packages in your project. List the packages under a ``packages:`` header:
+
+``multi-package.yaml``
+
+.. code:: yaml
+
+  packages:
+  - <path to package>
+  - <path to package>
+
+For each package, specify the relative path to the directory containing the ``daml.yaml`` file for that package.
+
+Example
+-------
+
+In this example, the Daml project contains two packages, ``package-Model`` and ``package-Logic``. The ``multi-package.yaml`` file at the root of the project contains relative paths to the Model and Logic packages.
+
+``multi-package.yaml``
+
+.. code:: yaml
+
+   packages:
+   - ./package-Logic
+   - ./package-Model
+
+Here's the resulting project tree:
+
+::
+
+   > tree
+   .
+   ├── multi-package.yaml
+   ├── package-Logic
+   │   ├── daml/...
+   │   └── daml.yaml
+   └── package-Model
+       ├── .daml/dist/package-Model-1.0.0.dar
+       ├── daml/...
+       └── daml.yaml
+
+With this configuration, running ``daml build`` on ``package-Logic`` also automatically rebuilds ``Model.dar`` as a data dependency of ``package-Model``.
+
+.. code:: bash
+
+   > # ... make changes to Model ...
+   > cd ./package-Logic/ # Navigate to package-Logic
+   > daml build # Build package Logic
+   ...
+   Dependency "package-Model" is stale, rebuilding...
+   ...
+   Building "package-Logic"...
+   Done.
+
+With multi-package builds configured, you can run ``daml build`` just once to build interdependent packages, with a guarantee that changes to dependencies are always propagated wherever they are needed.
+
+Build all packages in a project
+-------------------------------
+
+To build all packages in a project, use the ``--all`` flag. This
+flag builds every package listed in the specified ``multi-package.yaml``. 
+With the ``--all`` flag, you can run ``daml build`` from outside a package directory.
+
+.. _multi-package-yaml-location:
+
+How the CLI finds multi-package.yaml
+------------------------------------
+
+In most cases, you'll run ``daml build`` and ``daml clean`` from deeper in your project structure than the root. The discovery logic for these commands follows the patterns of ``daml.yaml``:
+
+1. If a path is specified with ``--multi-package-path PATH``, use 
+   the ``multi-package.yaml`` at that location. (This is the
+   ``multi-package`` equivalent to ``--project-root``.)
```

Comment:
If we have docs for `--project-root`, it'd be nice to link them here

Reply:
Yeah, it would :) Will have to be a follow-up task because I'm not finding anything useful